pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -643,7 +647,41 @@ func podAffinity(i *infinispanv1.Infinispan, matchLabels map[string]string) *cor
 	return i.Spec.Affinity
 }
 
-// deploymentForInfinispan returns an infinispan Deployment object
+func (r *ReconcileInfinispan) updatePodsLabels(m *infinispanv1.Infinispan, podList *corev1.PodList) error {
+	if len(podList.Items) == 0 {
+		return nil
+	}
+
+	labelsForPod := PodLabels(m.Name)
+	m.AddOperatorLabelsForPods(labelsForPod)
+	m.AddLabelsForPods(labelsForPod)
+
+	for _, pod := range podList.Items {
+		podLabels := make(map[string]string)
+		for index, value := range pod.Labels {
+			if _, ok := labelsForPod[index]; ok || consts.StatefulSetPodLabels[index] {
+				podLabels[index] = value
+			}
+		}
+		for index, value := range labelsForPod {
+			podLabels[index] = value
+		}
+
+		_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, &pod, func() error {
+			if pod.CreationTimestamp.IsZero() {
+				return errors.NewNotFound(corev1.Resource(""), pod.Name)
+			}
+			pod.Labels = podLabels
+			return nil
+		})
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+// statefulSetForInfinispan returns an infinispan StatefulSet object
 func (r *ReconcileInfinispan) statefulSetForInfinispan(m *infinispanv1.Infinispan, secret *corev1.Secret, configMap *corev1.ConfigMap) (*appsv1.StatefulSet, error) {
 	reqLogger := log.WithValues("Request.Namespace", m.Namespace, "Request.Name", m.Name)
 	lsPod := PodLabels(m.Name)
```

Comment:
@rigazilla  Why do we need `lsPod` to be distinct from `labelsForPod`? Won't both result in a match as all of the labels will be on the pods?

Reply:
The match will probably work, but I would not mix up internal labels `lsPod` with user labels `labelsForPod` which come for the user application domain and are not intended to be used as pods selector.